src/include/sof/lib_manager.h

```diff
@@ -81,7 +81,9 @@ struct ipc_lib_msg {
 struct ext_library {
 	struct k_spinlock lock;	/* last locking CPU record */
 	struct sof_man_fw_desc *desc[LIB_MANAGER_MAX_LIBS];
+#ifdef __SOF_MODULE_SERVICE_BUILD__
```

Comment:
This is first time this define is used in basefw build (this is defined when building modules with lmdk but not set for base fw build). I think this has to be a Kconfig for the main build to select loadable module behaviour and/or support both.

Reply:
@kv2019i right, yes, I agree that ideally we should be able to build firmware that can load all kinds of modules: IADK, native system-services, and native dynamic. But initially I'd integrate all three approaches into the firmware and then check how we can identify modules at run-time

Reply:
in fact indeed it isn't correct. `-D__SOF_MODULE_SERVICE_BUILD__=1` is only used to build the **module** itself, not the complete firmware. Converted to draft for a fix